### ISSUE TITLE
Unify and fix accessibility of the various Pagination components

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58133]
### Description

The editor uses a few user interfaces where pagination buttons are in place. For example:

- dataviews
- patterns explorer 
- global styles revisions
- possibly other places I may have missed 

While theoretically these pagination are very similar and should work in a very similar way, they are all different and use different components.

From an accessibility perspective, there are a few issues. Most notably:

- Buttons get dynamically disabled while they are focused, thus triggering a focus loss. See https://github.com/WordPress/gutenberg/issues/56547
- The buttons don't show a tooltip so that their accessible name isn't visually exposed.

I'd tend to think these pagination tools should _work and look_ the same way throughout the whole user interface.
It's also a good opportunity to fix the most relevant accessibility issuses and abstract some code.

Screenshots:

Dataviews pagination:

![01 pagination dataviews](https://github.com/WordPress/gutenberg/assets/1682452/45b1738e-2a5b-4c83-a2be-eb8dfc1f993e)

Patterns explorer pagination:

![02 pagination patterns explorer](https://github.com/WordPress/gutenberg/assets/1682452/d3932eeb-5bc0-4630-a83a-1673ffe72eda)

Global styles revisions pagination:

![03 revisions pagination](https://github.com/WordPress/gutenberg/assets/1682452/f9797f35-ec71-41d5-8a2d-5ebbf2cb2455)

NEW: Pages pagination: note: as far as I can tell this is the only case where the buttons do not trigger a focus loss because they use `arira-disabled` instead of `disabled`. Tooltips work as well. The implementation of the current page indicator is arguable though as it's a mix of translatable strings and labeling / value of the `<select>` element.

![new pages pagination](https://github.com/WordPress/gutenberg/assets/1682452/574de3cb-ca8e-4cb3-bec7-a320e1967deb)

### Step-by-step reproduction instructions

- Test one of these paginations for example the global styles one.
- Go to Site editor > Styles > Revisions.
- Use the keyboard and Tab to the 'Last page' button.
- Observe the button doesn't show a tooltip.
- Press Enter.
- Observe the 'Last page' button is now not focusable because it has a `disabled` HTML attribute.
- Observe focus is lost: press Tab again and observe the tab order starts from scratch from the document root.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by mikachan** [comment]
Noting that the Font Library modal also includes a pagination implementation, also discussed in https://github.com/WordPress/gutenberg/pull/59591.

![image](https://github.com/WordPress/gutenberg/assets/1645628/f2dccb1c-d07f-4c5e-9f33-151b2693ec37)

**Comment by t-hamano** [comment]
Related to this issue, I suggested adding the `Pagination` component.

https://github.com/WordPress/gutenberg/issues/55650
